rustls/tests/runners/kernel.rs

```diff
@@ -0,0 +1,22 @@
+#![allow(clippy::duplicate_mod)]
+
+#[macro_use]
+mod macros;
+
+#[cfg(feature = "ring")]
+#[path = "."]
+mod tests_with_ring {
+    provider_ring!();
+
+    #[path = "../kernel.rs"]
+    mod kernel;
+}
+
+#[cfg(feature = "aws_lc_rs")]
+#[path = "."]
+mod tests_with_aws_lc_rs {
+    provider_aws_lc_rs!();
+
+    #[path = "../kernel.rs"]
+    mod tests;
+}
```

Comment:
Can we do this without adding yet another small file?

Reply:
Sure. Where should I put the tests? I can stick them under `unbuffered`

Reply:
Yup, that sounds okay.